### PR TITLE
Milestone3

### PR DESCRIPTION
**Pull request by 300ms**
This pull request has been created for a code review. There is a misunderstanding about omniauth so please ignore changes about omniauth and omniauth-facebook.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def show
```
Comment (mosaaleb) [inline app/controllers/users_controller.rb:4]: - [x] user controller # index displays all users



Just implement the user index and ignore the friendship functionality until friendships milestone.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Post < ApplicationRecord
+  belongs_to :user
+  default_scope -> { order(created_at: :desc) }
+  validates :user_id, presence: true
```
Comment (mosaaleb) [inline app/models/post.rb]: - [x] `belongs_to :user` adds presence validation to `user_id` by default.
[check this](https://blog.bigbinary.com/2016/02/15/rails-5-makes-belong-to-association-required-by-default.html)

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  has_many :posts, dependent: :destroy
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
```
Comment (mosaaleb) [inline app/models/user.rb]: - [x] **[OPTIONAL]** devise already takes care of user attributes (email, password), if you checked `config/initializers/devise.rb` file, you will find all email and password configuration and you can edit it there.

File: app/views/posts/_post.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<li>
+    <p><%= post.title %></p>
+    <p><%= post.content %></p>
+    <h6>Updated at: <%= post.updated_at %></h6>
+    <% if @logged_in %>
```
Comment (mosaaleb) [inline app/views/posts/_post.html.erb]: - [x] There shoud be absolutely NO logic in views.





View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)
Rails has built-in mechanism to deal with view-related logic which is View Helpers (app/helpers/*_helper.rb). So the common convention is to move any logic in the views to the helpers. Please take a look at [this](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) for more information.

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<h1>Posts</h1>
+<%= link_to "Home", root_path %>
+<% if user_signed_in? %>
```
Comment (mosaaleb) [inline app/views/posts/index.html.erb]: - [x] Please move the view logic to post helper module.

File: app/views/posts/_post.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+        <h6><%= User.find_by(id: post.user_id).first_name %></h6>
```
Comment (mosaaleb) [inline app/views/posts/_post.html.erb]: - [x] Why not just `post.user.first_name`, we already have the right association in our models

**Review by mosaaleb — CHANGES_REQUESTED** [review]
## Well done Team :clap:

#### we still need to work on few requirements before moving on

- [x] our specs are empty



Let's add model and integration specs for `post` functionality.

- [x] check out my comments below.


#### Please handle these requirements and submit a new code review

##### Muhammad Ebeid

**Review by Ispirett — APPROVED** [review]
## Status Approved :+1:  :100: 
Great work implementing  the tests ask requested @300ms @WCanirinka 
Please  consider adding more tests for model functionality, like the downcase method for the user and
maximum length for content.

**Pull request merged**